components/pthread/include/esp_pthread.h

```diff
@@ -24,11 +27,33 @@ extern "C" {
 
 /** pthread configuration structure that influences pthread creation */
 typedef struct {
-    size_t stack_size;    ///< the stack size of the pthread
-    size_t prio;          ///< the thread's priority
-    bool inherit_cfg;     ///< inherit this configuration further
+    size_t stack_size;  ///< The stack size of the pthread
+    size_t prio;        ///< The thread's priority
+    bool inherit_cfg;   ///< Inherit this configuration further
+    // This member has to be an array and not a pointer due to the fact that the entire
+    // struct is copied by the thread when inherit_cfg is true which in turn likely
+    // will result in dangling pointers, especially if threads are created/destroyed
```

Comment:
Could we possibly use some FreeRTOS API to get the name from the parent task, instead of getting it from this structure? I.e. if inherit_cfg is true then make a copy of this structure and set thread_name pointer to NULL. When inheriting, if thread_name is NULL, get the name of the current task

Reply:
I see no API that can get the name of the parent API, no.

Reply:
Consider a case when task A creates task B which creates task C.
A sets inherit=true and provides a name (pointer), then creates B. At this point, B gets it's name from the pointer. Pthread config associated with B gets updated to keep inherit=true and name=NULL. When B creates C, we see that inherit=true but name=NULL, so we take the name of the current task (B). Thus C gets created with the name of B.

Reply:
Oh, I thought you meant there were an actual API to get the name of the creating task. I'll make a new commit with the proposed logic.